public/pages/DetectorDetail/containers/DetectorDetail.tsx

```diff
@@ -271,30 +274,27 @@ export const DetectorDetail = (props: DetectorDetailProps) => {
                 {detector &&
                 detector.enabled &&
                 detector.curState === DETECTOR_STATE.RUNNING ? (
-                  <EuiHealth color="success">
+                  <EuiHealth color={DETECTOR_STATE_COLOR.RUNNING}>
                     Running since{' '}
                     {detector.enabledTime
                       ? moment(detector.enabledTime).format('MM/DD/YY h:mm A')
                       : '?'}
                   </EuiHealth>
                 ) : detector.enabled &&
                   detector.curState === DETECTOR_STATE.INIT ? (
-                  <EuiHealth color="primary">Initializing</EuiHealth>
+                  <EuiHealth color={DETECTOR_STATE_COLOR.INIT}>Initializing</EuiHealth>
                 ) : detector.curState === DETECTOR_STATE.INIT_FAILURE ||
                   detector.curState === DETECTOR_STATE.UNEXPECTED_FAILURE ? (
-                  <EuiHealth color="danger">Initialization failure</EuiHealth>
-                ) : (
-                  <EuiHealth color="subdued">
-                    {detector.featureAttributes &&
-                    detector.featureAttributes.length
-                      ? detector.disabledTime
-                        ? `Stopped at ${moment(detector.disabledTime).format(
-                            'MM/DD/YY h:mm A'
-                          )}`
-                        : 'Detector is stopped'
-                      : 'Feature required to start the detector'}
+                  <EuiHealth color={DETECTOR_STATE_COLOR.INIT_FAILURE}>Initialization failure</EuiHealth>
+                ) : detector.curState === DETECTOR_STATE.DISABLED ? (
+                  <EuiHealth color={DETECTOR_STATE_COLOR.DISABLED}>
+                    {detector.disabledTime
+                    ? `Stopped at ${moment(detector.disabledTime).format('MM/DD/YY h:mm A')}`
+                    : 'Detector is stopped'}
                   </EuiHealth>
-                )}
+                ) : detector.curState === DETECTOR_STATE.FEATURE_REQUIRED ? (
+                  <EuiHealth color={DETECTOR_STATE_COLOR.FEATURE_REQUIRED}>Feature required to start the detector</EuiHealth>
+                ) : isLoading ? 'Loading...' : ''}
```

Comment:
one thought: what about using some animation of [EUI](https://elastic.github.io/eui/#/display/loading), such as **Text content**, instead of 'Loading...'? Maybe you can confirm with UX on this.

Reply:
I agree, that could be cleaner. I will change.